### PR TITLE
Extra db enum constancy checks 

### PR DESCRIPTION
**Pull request by rustyrussell**
This is @renepickhardt 's #1764 rebased and with my braindamage fixed.

Closes: #1764

### CODE REVIEW

File: wallet/wallet.c [hunk 1]
```diff
@@ -161,33 +160,40 @@ bool wallet_update_output_status(struct wallet *w,
...
-		sqlite3_bind_int(stmt, 1, newstatus);
+		sqlite3_bind_int(stmt, 1, output_status_in_db(newstatus));
 		sqlite3_bind_blob(stmt, 2, txid, sizeof(*txid), SQLITE_TRANSIENT);
 		sqlite3_bind_int(stmt, 3, outnum);
 	}
 	db_exec_prepared(w->db, stmt);
 	return sqlite3_changes(w->db->sql) > 0;
 }
 
+#define SELECT_UTXO							\
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply wallet/wallet.c]: Agreed, changed.

**Comment by rustyrussell** [comment]
The fatal() is because it's possible that a value is not a valid enum: in C, they're ints.

The error is because I didn't update run-db.c to use the db_fatal hook.... fixing...

**Comment by cdecker** [comment]
> The fatal() is because it's possible that a value is not a valid enum: in C, they're ints.

I know, I was wondering whether the compiler would be smart enough to optimize it away :wink:

**Comment by cdecker** [comment]
ACK cd3ae79

File: wallet/wallet.h [hunk 2]
```diff
@@ -55,10 +59,31 @@ enum output_status {
 	output_state_any = 255
 };
 
+static inline enum output_status output_status_in_db(enum output_status s)
+{
+	switch (s) {
```
Comment (ZmnSCPxj) [inline wallet/wallet.h]: The  enum definition has a `/!\` warning sign, but, why not here also?

Although I suppose someone who decides to change the numbering, even if, there exists already `/!\` in the enum declaration, would not be stopped, by a `/!\` here.

**Pull request merged**